### ISSUE TITLE
Please explain why an environment is required

### ISSUE THREAD

**Issue by megamorf** [actions/deploy-pages#85]
Hey everyone, I'm working for a big user of GHES and there are some policies in place to limit the use of GH actions to official/verified actions and custom-built ones. 

Since that means we'll have to rely on this action to publish to GitHub Pages I had a look at the README. However, it is not clear to me why an environment is needed here at all. Even being required to have an artifact before (rather than also supporting the build output of previous steps) makes the entire process strangely complex.

Could you please explain your reasoning behind this and update the docs accordingly, thanks!

**Comment by yoannchaudet** [comment]
TL;DR: security.

Long answer:

Traditionally Pages has been published from a source branch in a repository. Users have been able to restrict who can publish a Pages sites (even with write access on a repository) using [branch protection rules](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/defining-the-mergeability-of-pull-requests/managing-a-branch-protection-rule).

When building and publishing a Pages site from a workflow, we explicitly removed this "source" branch limitation. It does not make sense to force the workflow to run from a specific branch when there are so many viable [events](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows) which could trigger a workflow. We not only require an environment to be specified when doing a deployment, we require it to be [protected](https://docs.github.com/en/actions/deployment/targeting-different-environments/using-environments-for-deployment#environment-protection-rules). This requirement is simply our security mechanism for allowing users to have a finer control over Pages deployments like they always had with source branches. In addition, specifying an environment means we automatically create [deployments](https://docs.github.com/en/rest/deployments/deployments?apiVersion=2022-11-28) on your behalf. A lot of investment are being made on that front currently.
